ros2param/ros2param/api/__init__.py

```diff
@@ -13,113 +13,45 @@
-    """Guess the desired type of the parameter based on the string value."""
-    value = ParameterValue()
+    value = None
     try:
-        yaml_value = yaml.safe_load(string_value)
-    except yaml.parser.ParserError:
-        yaml_value = string_value
-
-    if isinstance(yaml_value, bool):
-        value.type = ParameterType.PARAMETER_BOOL
-        value.bool_value = yaml_value
-    elif isinstance(yaml_value, int):
-        value.type = ParameterType.PARAMETER_INTEGER
-        value.integer_value = yaml_value
-    elif isinstance(yaml_value, float):
-        value.type = ParameterType.PARAMETER_DOUBLE
-        value.double_value = yaml_value
-    elif isinstance(yaml_value, list):
-        if all((isinstance(v, bool) for v in yaml_value)):
-            value.type = ParameterType.PARAMETER_BOOL_ARRAY
-            value.bool_array_value = yaml_value
-        elif all((isinstance(v, int) for v in yaml_value)):
-            value.type = ParameterType.PARAMETER_INTEGER_ARRAY
-            value.integer_array_value = yaml_value
-        elif all((isinstance(v, float) for v in yaml_value)):
-            value.type = ParameterType.PARAMETER_DOUBLE_ARRAY
-            value.double_array_value = yaml_value
-        elif all((isinstance(v, str) for v in yaml_value)):
-            value.type = ParameterType.PARAMETER_STRING_ARRAY
-            value.string_array_value = yaml_value
-        else:
-            value.type = ParameterType.PARAMETER_STRING
-            value.string_value = string_value
-    else:
-        value.type = ParameterType.PARAMETER_STRING
-        value.string_value = yaml_value
+        value = parameter_value_to_python(parameter_value)
+    except RuntimeError as e:
+        print(f'Runtime error {e} raised')
     return value
 
 
-def parse_parameter_dict(*, namespace, parameter_dict):
-    parameters = []
-    for param_name, param_value in parameter_dict.items():
-        full_param_name = namespace + param_name
-        # Unroll nested parameters
-        if type(param_value) == dict:
-            parameters += parse_parameter_dict(
-                    namespace=full_param_name + PARAMETER_SEPARATOR_STRING,
-                    parameter_dict=param_value)
-        else:
-            parameter = Parameter()
-            parameter.name = full_param_name
-            parameter.value = get_parameter_value(string_value=str(param_value))
-            parameters.append(parameter)
-    return parameters
-
-
-def load_parameter_dict(*, node, node_name, parameter_dict):
-
-    parameters = parse_parameter_dict(namespace='', parameter_dict=parameter_dict)
-    response = call_set_parameters(
-        node=node, node_name=node_name, parameters=parameters)
+def load_parameter_file(*, node, node_name, parameter_file, use_wildcard):
+    # Remove leading slash and namespaces
+    client = AsyncParameterClient(node, node_name)
+    client.wait_for_services(timeout_sec=5.0)
+    if not client.services_are_ready():
```

Comment:
I think this is equivalent:
```suggestion
    if not client.wait_for_services(timeout_sec=5.0):
```